src/scripts/underground/Mine.ts

```diff
@@ -35,30 +33,60 @@ class Mine {
         Mine.grid = tmpGrid;
         Mine.rewardGrid = tmpRewardGrid;
 
+        let Diff = 0;
+        let Added = 0;
         for (let i = 0; i < App.game.underground.getMaxItems(); i++) {
             const item = UndergroundItem.getRandomItem();
-            const x = Mine.getRandomCoord(this.sizeX, item.space[0].length);
-            const y = Mine.getRandomCoord(this.sizeY, item.space.length);
+            const x = Mine.getRandomCoord(Underground.sizeX, item.space[0].length);
+            const y = Mine.getRandomCoord(App.game.underground.getNewYLayer(), item.space.length);
             const res = Mine.canAddReward(x, y, item);
             if (res) {
                 Mine.addReward(x, y, item);
+                Added = Added + 1;
             }
         }
+
+        if (App.game.underground.getMaxItems() < App.game.underground.getMinItems()) {
+            Diff = App.game.underground.getMinItems() - App.game.underground.getMaxItems();
+            //Late-Game players might do some cheesy shit and upgrade min to be above their max. This makes sure it deducts the difference.
+        }
+        for (Added; Added < App.game.underground.getMinItems() - Diff;) {
```

Comment:
Could this be simplified with something like:
```suggestion
        // Check in case player upgrade min above max
        const min = Math.min(App.game.underground.getMinItems(), App.game.underground.getMaxItems());
        while (added < min) {
```

Can you also change `Added` to be lowercased as it's not a global value.
And remove `Diff` if the above code works as you intended.